File: SchoolBack/src/main/kotlin/com/open_inc/SchoolBack/models/Review.kt
package com.open_inc.SchoolBack.models

import jakarta.persistence.*

@Entity
@Table(name = "Review")
data class Review(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,

    @ManyToOne
    @JoinColumn(name = "user_id", nullable = false)
    val user: User,

    @Column(name = "rate", nullable = false)
    val rate: Int = 0,

    @Column(name = "text", length = 1024, nullable = false)
    val text: String = "",

    @Column(name="source", length = 512, nullable = true)
    val source:String = "",

    @Column(name="show_name")
    val showUserData:Boolean=true


)

File: SchoolFront/lib/dataclasses/review.dart
class Review{
  final int rate;
  final String text;
  String source="";
  String userData="Аноним";
  bool showUserData;

  Review({
    required this.rate,
    required this.text,
    this.source='',
    this.userData='',
    this.showUserData=false
  });

  Review.undefined():
      rate=0, text='', userData='', source='', showUserData=true;

  Map<String, dynamic> getJson(){
    return {
      "rate": this.rate,
      "text": this.text,
      "source": this.source,
      "showUserData": this.showUserData
    };
  }
}